calculadora_do_cidadao/fields.py

```diff
@@ -14,6 +15,20 @@ def deserialize(cls, value: str) -> Decimal:
 
 
 class DateField(fields.DateField):
-    """Field for mmm/yyyy dates."""
+    """DateField which supports different date formats, including Brazilian"""
 
-    INPUT_FORMAT = "%b/%Y"
+    INPUT_FORMATS = ("%Y-%m-%d", "%d/%m/%Y", "%Y-%m", "%m/%Y", "%b/%Y", "%Y")
+
+    @classmethod
+    def deserialize(cls, value, *args, **kwargs):
+        value = fields.Field.deserialize(value)
+        value = fields.as_string(value)
+
+        for date_format in cls.INPUT_FORMATS:
+            try:
+                dt_object = datetime.datetime.strptime(value, date_format)
+            except ValueError:
+                continue
+            else:
+                return datetime.date(dt_object.year, dt_object.month, dt_object.day)
```

Comment:
Uma sugestão e uma pergunta.

Pergunta: Porque usar `try/except/else` ao invés de só `try/except`? Algum motivo específico?

Sugestão: não precisa parsear a data novamente, pode só retornar `dt_object.date()` :)

```suggestion
            try:
                dt_object = datetime.datetime.strptime(value, date_format)
            except ValueError:
                continue
            return dt_object.date()
```

Reply:
Opa, obrigado pela sugestão! Não conhecia esse método (e [já até mudei na rows](https://github.com/turicas/rows/commit/42bbe4ac79b2ebcee47c400aa40847b24e63f407) - ficou bem mais rápido).
Prefiro usar o `else` (apesar de ser desnecessário) apenas para explicitar que ou vai executar o `continue`, ou vai executar o código do `else`.